.github/workflows/benchmark.yml

```diff
@@ -5,11 +5,15 @@ on:
   push:
     branches:
       - "main"
+  schedule:
+    - cron: '0 0 * * *'
 
 jobs:
-
   run_benchmarks:
-    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        os: ['ubuntu-18.04', 'ubuntu-20.04', 'ubuntu-22.04', 'windows-2016', 'windows-2019', 'windows-2022']
```

Comment:
Do we need to run on different os? Is it possible / likely that we'll catch a regression on only one os. I'm not sure django has much os specific code paths. 

We need to consider the environmental impact of running all these multiple versions and the benefit we will get from it.

Reply:
I have decreased the number of machines to just one for ubuntu and one for windows. I viewed the results by running the benchmarks on these two machines, there was some difference in performance between them.